horace_core/sqw/@Experiment/private/check_sample_or_inst_array_and_return_std_form_.m

```diff
@@ -42,6 +64,13 @@
             'must be inst or sample but some elements of the input cellarray are not');
     end
     std_form = std_form.add(sample_or_instrument);
+elseif isa(sample_or_instrument,'unique_objects_container');
+    is = strcmp( sample_or_instrument.baseclass, std_form.baseclass);
+    if ~all(is)
+        error('HORACE:Experiment:invalid_argument', ...
+            'must be inst or sample but the input container is not');
+    end
```

Comment:
Isn't this check handled by add anyway? Wouldn't it be easier to let it happen there?

Reply:
The internal check would be on a per-element basis. This check sets it all out up front for a whole input to ensure it is sane. It fits in with the rest of the checks here. I think we'll leave it.

Reply:
My argument would be that the fewer places the checks happen, the fewer checks that have to change should this change. We should be able to trust in the uoc to handle the input.